base/math.jl

```diff
@@ -698,7 +698,9 @@ end
 @inline ^(x::Float64, y::Integer) = x ^ Float64(y)
 @inline ^(x::Float32, y::Integer) = x ^ Float32(y)
 @inline ^(x::Float16, y::Integer) = Float16(Float32(x) ^ Float32(y))
-@inline ^{p}(x::Float16, ::Type{Val{p}}) = Float16(Float32(x) ^ Val{p})
+@inline ^{p}(x::Float16, ::Type{Val{p}}) = Float16(x ^ p)
+@inline ^{p}(x::Float32, ::Type{Val{p}}) = Float32(x ^ p)
+@inline ^{p}(x::Float64, ::Type{Val{p}}) = Float64(x ^ p)
 
```

Comment:
I've removed the methods I added and put in the `literal_pow` one and get:
```julia
Test threw an exception of type MethodError
  Expression: x ^ Val{1} === x
  MethodError: no method matching ^(::Float32, ::Type{Val{1}})
  Closest candidates are:
    ^(::Float32, !Matched::Float32) at math.jl:697
    ^(::Float32, !Matched::Integer) at math.jl:699
    ^(!Matched::BigInt, ::Type{Val{p}}) where p at gmp.jl:447
```

Do I need to add a line for the `Float32` and `Float64` types similar to `Float16`?